airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination/record_buffer/BufferingStrategy.java

```diff
@@ -20,14 +21,16 @@
 public interface BufferingStrategy extends AutoCloseable {
 
   /**
-   * Add a new message to the buffer while consuming streams
+   * Add a new message to the buffer while consuming streams, also handles when a buffer flush when
+   * buffer has been filled
    *
-   * @param stream - stream associated with record
-   * @param message - {@link AirbyteMessage} to buffer
-   * @return true if this record cause ALL records in the buffer to flush, otherwise false.
+   * @param stream stream associated with record
+   * @param message {@link AirbyteMessage} to be added to the buffer
+   * @return an optional value if a flushed occur with the respective flush type, otherwise an empty
+   *         value means only a record was added
    * @throws Exception throw on failure
    */
-  boolean addRecord(AirbyteStreamNameNamespacePair stream, AirbyteMessage message) throws Exception;
+  Optional addRecord(AirbyteStreamNameNamespacePair stream, AirbyteMessage message) throws Exception;
```

Comment:
Also, may have to change this to `Optional<BufferFlushType>` to appease SpotBugs